core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -2962,4 +2970,23 @@ assert allLessThan(this.oldToNewOutputs.values(),
     /** Sets {@link #decorrelator}. */
     Config withDecorrelator(RelDecorrelator decorrelator);
   }
+
+  // -------------------------------------------------------------------------
+  //  Getter/Setter
+  // -------------------------------------------------------------------------
+
+  /**
+   * Returns the {@code visitor} on which the {@code MethodDispatcher} dispatches
+   * each {@code decorrelateRel} method, the default implementation returns this instance,
+   * if you got a sub-class, override this method to replace the {@code visitor} as the
+   * sub-class instance.
+   */
+  protected RelDecorrelator getVisitor() {
+    return this;
+  }
+
+  /** Returns the rules applied on the rel after decorrelation, never null. */
+  protected Collection<RelOptRule> getPostDecorrelateRules() {
```

Comment:
Do we want to consider annotating it with "@NotNull"?

Reply:
No, because `@NotNull` should be the default.